src/components/icon/icon-registry.ts

```diff
@@ -0,0 +1,239 @@
+import {Injectable} from 'angular2/core';
+import {Http} from 'angular2/http';
+import {Observable} from 'rxjs/Rx';
+
+/**
+  *  Configuration for an icon, including the URL and possibly the cached SVG element.
+  */
+class IconConfig {
+  svgElement: SVGElement = null;
+  constructor(public url: string) {
+  }
+}
+
+@Injectable()
+export class MdIconRegistry {
+  // IconConfig objects and cached SVG elements for individual icons.
+  // First level of cache is namespace (which is the empty string if not specified).
+  // Second level is the icon name within the namespace.
+  private _iconConfigs = new Map<string, Map<string, IconConfig>>();
```

Comment:
My preference would be for flattening the icon namespace + name into a single key, rather than doing a map-of-maps. @hansl do you have an opinion here?